onadata/libs/tests/test_permissions.py

```diff
@@ -128,6 +128,28 @@ def test_get_object_users_with_permission(self):
         self.assertIn('metadata', users_with_perms_first_keys)
         self.assertIn('is_org', users_with_perms_first_keys)
 
+    # pylint: disable=C0103
+    def test_user_profile_exists_for_users_with_perms(self):
+        """
+        Test user profile exists when retrieving users with perms
+        """
+        alice = self._create_user('alice', 'alice')
+        # do not manually create user profile for alice, should be
+        # handled by get_object_users_with_permissions()
+        org_user = tools.create_organization("modilabs", alice).user
+        demo_grp = Group.objects.create(name='demo')
+        alice.groups.add(demo_grp)
+        self._publish_transportation_form()
+        EditorRole.add(org_user, self.xform)
+        EditorRole.add(demo_grp, self.xform)
+        users_with_perms = get_object_users_with_permissions(
+            self.xform, with_group_users=True)
+        self.assertTrue(org_user in [d['user'] for d in users_with_perms])
+        self.assertTrue(alice in [d['user'] for d in users_with_perms])
+        for d in users_with_perms:
+            user_obj = d['user']
+            self.assertTrue(hasattr(user_obj, "profile"))
```

Comment:
Is there a check for a user account that does not have the profile created? I believe the test should verify the profile is created when a user does not have a user profile. 

```python
self.assertFalse(hasattr(user_obj, "profile")
users_with_perms = get_object_users_with_permissions(
            self.xform, with_group_users=True)
self.assertTrue(hasattr(user_obj, "profile"))
```

Reply:
@ukanga Added a post_signal [here](https://github.com/onaio/onadata/pull/2344/commits/a3ac18965566a286cd964b9ab8d15f45600b3931) so the missing profile check might not be necessary

Reply:
Then perhaps we do not need this test since the function is already tested. We might still need the get_or_create in https://github.com/onaio/onadata/blob/c8349e030b913ce7eca44945134c1618911417da/onadata/libs/serializers/project_serializer.py#L179 since we may have users without profiles in some deployments. The signal should handle new user creations but does not handle the backlog.